app/controllers/api/v1/campaigns_controller.rb

```diff
@@ -0,0 +1,66 @@
+class Api::V1::CampaignsController < ApplicationController
+  def show
+    @id = params[:id]
+    @campaign = Campaign.find(@id)
+    audience_label_ids = @campaign.audience.pluck('id')
+    contact_counts = Contact.joins(:taggings).where(taggings: { tag_id: audience_label_ids }).distinct.count
+    formatted_campaign = {
+      id: @campaign.id,
+      title: @campaign.title,
+      message: @campaign.message,
+      campaign_type: @campaign.campaign_type,
+      campaign_status: @campaign.campaign_status,
+      audience: @campaign.audience,
+      scheduled_at: @campaign.scheduled_at,
+      sent_count: @campaign.sent_count,
+      template: @campaign.template,
+      delivered_count: @campaign.delivered_count,
+      failed_count: @campaign.failed_count,
+      clicked_count: @campaign.clicked_count,
+      read_count: @campaign.read_count,
+      replied_count: @campaign.replied_count,
+      audience_count: contact_counts
+    }
```

Comment:
Why not simply do @campaign.audience_count = contact_counts instead of declaring a new variable. Also does campaign need to be an instance variable ?